### PR TITLE
[DRAFT] webpack update

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Update our build process for ORA2 JS to use webpack, thus allowing use of ES6 tools/syntax.
Uses frontend-build in order to closely align with MFE build practices.

**What changed?**
o.0

no more requireJS imports.
All JS is imported/exported from files.
npm configuration files for webpack builds based on frontend-build.
webpack build commands in Makefile.

ESlint styling changes, where feasible/easy/automatic.
Note: There are a number of remaining issues that are currently ignored by disabling rules in the eslint config that ABSOLUTELY should be addressed in the future (especially number of classes exported from a given file).

**Developer Checklist**
- [ ] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [ ] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [JIRA-5190](https://openedx.atlassian.net/browse/JIRA-5190)

FIY: @edx/masters-devs-gta

### CODE REVIEW

**Review by jansenk — COMMENTED** [review]
I've read through the changes and I have no concerns

File: karma.conf.js [hunk 1]
```diff
@@ -105,12 +101,27 @@ module.exports = function(config) {
 
     // start these browsers
     // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-    browsers: ['PhantomJS'],
+    browsers: ['HeadlessChrome'],
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply karma.conf.js:104]: clearing warnings.  PhantomJS not supported any more if I recall

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 2]
```diff
@@ -585,394 +578,403 @@ OpenAssessment.ResponseView.prototype = {
...
-    /**
+        this.updateFilesDescriptionsFields(files, descriptions, uploadType);
+      }
+
+      if (this.files === null) {
+        $(this.element).find('.file__upload').prop('disabled', true);
+      }
+    }
+
+   isUploadSupported = (file, uploadType) => {
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/lms/oa_response.js:640]: Is there a particular reason to move this to the bottom?

File: openassessment/xblock/static/js/src/studio/oa_edit_assessment.js [hunk 3]
```diff
@@ -449,105 +461,105 @@ OpenAssessment.EditStudentTrainingView.prototype = {
                      {
                          criterion: "brevity",
                          option: "suberb"
-                     },
+                     }
```
Comment (nsprenkle) [inline openassessment/xblock/static/js/src/studio/oa_edit_assessment.js:464]: Overeager search and replace?

**Review by nsprenkle — COMMENTED** [review]
Overall looks good, pending continued successful smoke test and removal of `console.log`s. Random questions and comments (largely for my own curiosity) peppered throughout.

File: webpack.dev.config.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }),
+    new MiniCssExtractPlugin({
+      filename: '[name].css',
+    })
+  ],
+});
+
+config.resolve.modules = ['node_modules'].concat(
+  // UPDATE ME
+  path.resolve(__dirname, 'openassessment/xblock/static/js/src')
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply webpack.dev.config.js:34]: lol, nothing.  This is from the conversion template.  will remove, thx.

File: openassessment/xblock/studio_mixin.py [hunk 5]
```diff
@@ -81,14 +80,11 @@ def studio_view(self, context=None):  # pylint: disable=unused-argument
             self.STUDIO_EDITING_TEMPLATE
         ).render(self.editor_context())
         fragment = Fragment(rendered_template)
-        if settings.DEBUG:
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply openassessment/xblock/studio_mixin.py:84]: yes and no.  From what I understand from talking to people, the debug settings weren't really being exercised.  With the new architecture, a dev build is theoretically possible, which would use a different webpack config and different env variables, but it wouldn't function in the way this old code dictated.
The python would either be agnostic to the change or it would have to somehow be listening to an active webpack dev server

File: openassessment/xblock/static/js/src/lms/oa_response.js (hunk 2 above)
Comment (muselesscreator) [reply openassessment/xblock/static/js/src/lms/oa_response.js:640]: moving from a local function to a class method.
